consensus/hotstuff/packer/packer.go

```diff
@@ -42,8 +42,11 @@ func (p *SigDataPacker) Decode(data []byte) (*SignatureData, error) {
 	return &sigData, err
 }
 
-// UnpackRandomBeaconSig takes sigData previously packed by packer, decodes it and extracts random beacon signature
-func UnpackRandomBeaconSig(sigData []byte) (crypto.Signature, error) {
+// UnpackRandomBeaconSigForV2 takes sigData previously packed by packer,
+// decodes it and extracts random beacon signature
+// TODO: to be removed in V3. Remove it when the combined vote processor
```

Comment:
I don't understand this, in V3 we will use exactly the same code. Difference between V2 and V3 is in `AggregatedRandomBeaconSig`. But this unpacking is for `ReconstructedRandomBeaconSig`. We can't use packer here since it needs a committee, we only need to extract the random beacon sig so no point in doing full decoding of QC sig.

Reply:
Yes, you are right. We only need to read the random beacon sig. Good catch